litestar/handlers/websocket_handlers/listener.py

```diff
@@ -0,0 +1,234 @@
+        signature_namespace: Mapping[str, Any] | None = None,
+        type_encoders: TypeEncodersMap | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize ``WebsocketRouteHandler``
+
+        Args:
+            path: A path fragment for the route handler function or a sequence of path fragments. If not given defaults
+                to ``/``
+            dependencies: A string keyed mapping of dependency :class:`Provider <.di.Provide>` instances.
+            dto: :class:`DTOInterface <.dto.interface.DTOInterface>` to use for (de)serializing and
+                validation of request data.
+            exception_handlers: A mapping of status codes and/or exception types to handler functions.
+            guards: A sequence of :class:`Guard <.types.Guard>` callables.
+            middleware: A sequence of :class:`Middleware <.types.Middleware>`.
+            receive_mode: Websocket mode to receive data in, either `text` or `binary`.
+            send_mode: Websocket mode to receive data in, either `text` or `binary`.
+            name: A string identifying the route handler.
+            on_accept: Callback invoked after a connection has been accepted, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            on_disconnect: Callback invoked after a connection has been closed, receiving the
+                :class:`WebSocket <.connection.WebSocket>` instance as its only argument
+            opt: A string keyed mapping of arbitrary values that can be accessed in :class:`Guards <.types.Guard>` or
+                wherever you have access to :class:`Request <.connection.Request>` or
+                :class:`ASGI Scope <.types.Scope>`.
+            return_dto: :class:`DTOInterface <.dto.interface.DTOInterface>` to use for serializing
+                outbound response data.
+            signature_namespace: A mapping of names to types for use in forward reference resolution during signature
+                modelling.
+            type_encoders: A mapping of types to callables that transform them into types supported for serialization.
+            **kwargs: Any additional kwarg - will be set in the opt dictionary.
+        """
+        self._listener_context = _ListenerContext()
+        self._receive_mode: WebSocketMode = receive_mode
+        self._send_mode: WebSocketMode = send_mode
+        self._on_accept = AsyncCallable(on_accept) if on_accept else None
+        self._on_disconnect = AsyncCallable(on_disconnect) if on_disconnect else None
+        self.type_encoders = type_encoders
+
+        super().__init__(
+            path=path,
+            dependencies=dependencies,
+            exception_handlers=exception_handlers,
+            guards=guards,
+            middleware=middleware,
+            name=name,
+            opt=opt,
+            signature_namespace=signature_namespace,
+            dto=dto,
+            return_dto=return_dto,
```

Comment:
I don't think the DTOs should be passed to `WebsocketRouteHandler`, since it can't do anything with those.

Reply:
Well it does result in them being assigned to `self` - which is all any handler type does really, as any further use is done via `resolve_dto()`/`resolve_return_dto()`.

Reply:
I see your point though - it is confusing defining them in `WebsockeRouteHandler` kwargs - I'll remove.

Reply:
Hmm. You have a point there. I'm just thinking that, from a user's perspective it might be confusing to be able to pass a dto and then that not having any effect at all.

Reply:
I've come around to your PoV :)